public/components/datasources/components/manage/accelerations/create_accelerations_flyout/selectors/source_selector.tsx

```diff
@@ -0,0 +1,270 @@
+        setTables(tableLabels);
+      } else if (
+        (dbCache.status === CachedDataSourceStatus.Updated && dbCache.tables.length === 0) ||
+        dbCache.status === CachedDataSourceStatus.Empty
+      ) {
+        setTables([]);
+      }
+      setSelectedTable([]);
+      setAccelerationFormData({ ...accelerationFormData, dataTable: '' });
+      setLoadingComboBoxes({ ...loadingComboBoxes, dataTable: false });
+    }
+  };
+
+  useEffect(() => {
+    loadDataSource();
+  }, []);
+
+  useEffect(() => {
+    if (accelerationFormData.dataSource !== '') {
+      loadDatabases();
+    }
+  }, [accelerationFormData.dataSource]);
+
+  useEffect(() => {
+    if (accelerationFormData.database !== '') {
+      loadTables();
+    }
+  }, [accelerationFormData.database]);
+
+  return (
+    <>
+      <EuiText data-test-subj="datasource-selector-header">
+        <h3>Select data source</h3>
+      </EuiText>
+      <EuiSpacer size="s" />
+      <EuiText size="s" color="subdued">
+        Select the data source to accelerate data from. External data sources may take time to load.
+      </EuiText>
+      <EuiSpacer size="m" />
+
+      {dataSourcesPreselected ? (
+        <>
+          <EuiFlexGroup>
+            <EuiFlexItem>{dataSourceDescription}</EuiFlexItem>
+            <EuiFlexItem>
+              <EuiDescriptionList>
+                <EuiDescriptionListTitle>Database</EuiDescriptionListTitle>
+                <EuiDescriptionListDescription>
+                  {accelerationFormData.database}
+                </EuiDescriptionListDescription>
+              </EuiDescriptionList>
+            </EuiFlexItem>
+            <EuiFlexItem>
+              <EuiDescriptionList>
+                <EuiDescriptionListTitle>Table</EuiDescriptionListTitle>
+                <EuiDescriptionListDescription>
+                  {accelerationFormData.dataTable}
+                </EuiDescriptionListDescription>
+              </EuiDescriptionList>
+            </EuiFlexItem>
+          </EuiFlexGroup>
+        </>
+      ) : (
+        <>
+          {dataSourceDescription}
+          <EuiSpacer size="m" />
+          <EuiFormRow
+            label="Database"
+            helpText="Select the database that contains the tables you'd like to use."
+            isInvalid={hasError(accelerationFormData.formErrors, 'databaseError')}
+            error={accelerationFormData.formErrors.databaseError}
+          >
+            <EuiFlexGroup gutterSize="s">
+              <EuiFlexItem>
+                <EuiComboBox
+                  placeholder="Select a database"
+                  singleSelection={{ asPlainText: true }}
+                  options={databases}
+                  selectedOptions={selectedDatabase}
+                  onChange={(databaseOptions) => {
+                    if (databaseOptions.length > 0) {
+                      setAccelerationFormData(
+                        producer((accData) => {
```

Comment:
Is producer here being used to create a new object from acceleration data? If so, why is it needed as opposed to mutating accData?

Reply:
accData/AccelerationFormData contains large number of objects as it contains whole forms data. Hence, we rely on `immer` library provided `producer` function to update the state. This helps to do away with complex nesting stuff.

Reply:
Sounds good